server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/redirect/MailModifier.java

```diff
@@ -142,7 +146,7 @@ public void setSubjectPrefix(Mail originalMail) throws MessagingException {
     public void setReplyTo(Optional<MailAddress> optionalReplyTo) throws MessagingException {
         if (optionalReplyTo.isPresent()) {
             MailAddress replyTo = optionalReplyTo.get();
-            if (replyTo.equals(SpecialAddress.NULL)) {
+            if (replyTo.isNullSender() || replyTo.equals(SpecialAddress.NULL)) {
```

Comment:
doesn't make sense to me. Can we have methods with relevant names for such action ? I don't know what `replyTo.equals(SpecialAddress.NULL)` is and why it's different from `isNullSender`

Reply:
Well, I must confess my understanding of the entire `redirect` package is not good at all.

It breaks everywhere as soon as you do a single change, and in random places... I don't even understand what these special addresses are...

So basically I ask myself the same question than you...